### ISSUE TITLE
GTK3 - AltGr key detected as Ctrl key

### ISSUE THREAD

**Issue by giuspen** [msys2/MINGW-packages#13372]
### Description

In a GTK3 TextView, If you push 'AltGr', it is detected as a 'Ctrl'

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

Version	10.0.19044 Build 19044

### MINGW environments affected

- [X] MINGW64
- [ ] MINGW32
- [ ] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

### Expected behavior

AltGr down does not raise the flag GDK_CONTROL_MASK

### Actual behavior

AltGr raises the flag GDK_CONTROL_MASK

### Repro steps

Just build and run this small example that shows the issue https://github.com/giuspen/cherrytree/blob/master/sandbox/gtkmm/simpletextview.c

**Comment by lb90** [comment]
Hello, @giuspen! Thanks for reporting this!

Actually on Windows AltGr is the same as Ctrl + Alt. See for example https://devblogs.microsoft.com/oldnewthing/20120423-00/?p=7803:

> The example given is the famous[ AltGr modifier](https://devblogs.microsoft.com/oldnewthing/20040329-00/?p=40003), equivalent to Ctrl+Alt

Here are a few links where this is explained in more depth:

* https://en.wikipedia.org/wiki/AltGr_key#Ctrl+Alt
* https://github.com/microsoft/vscode/issues/50341

**Comment by giuspen** [comment]
Hi @lb90 many thanks for explaining this to me, I'm a bit ashamed I have never known it

**Comment by lb90** [comment]
Ah you're in good company! 😄 I didn't know that as well until very recently!

**Comment by dardoor** [comment]
AltGr and Ctrl+Alt are _not_ the same thing and it's certainly possible to differentiate between them in Windows.
An example was posted in this very thread: Microsoft's own VS Code.
Another example is Notepad++.

Both of those distinguish between Ctrl+Alt+key and AltGr+key presses and I think that's the proper way, because it gives you more options.

**Comment by lb90** [comment]
Thanks, @dardoor! Yes, it's possible to distinguish AltGr and Ctrl + Alt, I believe that GDK used ti di that before the rewrite in https://gitlab.gnome.org/GNOME/gtk/-/merge_requests/1051. Anyway I'm not sure if it's worth making those distinct, as it's a platform convention that shortcuts should match the same way. I'd like to hear more from app developers in that regard, as I'm not much knowledgeable in keyboard layouts and related things

**Comment by dardoor** [comment]
Yes, in fact @giuspen's CherryTree used to distinguish them before that change.

In my experience with using Windows, apps tend to either distinguish them (like Notepad++) or they prioritize one behavior over the other.
For example, LibreOffice prioritizes the input of special characters, so if you press either Ctrl+Alt or AltGr you get a special character instead of the assigned Ctrl+Alt shortcut. But some other apps prioritize app shortcuts, making it impossible to type special characters with AltGr.

Both of those are bad IMO and I would prefer to distinguish between them. But I can understand if someone might disagree.
There's also the _Ctrl+AltGr_ problem. IMO it should behave as Ctrl+Alt, but I don't know if that's possible.

Microsoft recommends to just not use Ctrl+Alt shortcuts in apps, which is more of a workaround than a solution.

I think a lot of developers are not aware of this mess, also because the the US keyboard layout doesn't have AltGr.

**Comment by lmattcris** [comment]
>Anyway I'm not sure if it's worth making those distinct, as it's a platform convention that shortcuts should match the same way.

Well, I hope it's _worth_ considering it, as on many localized keyboards the AltGr **is** a distinct key for composing characters in a **standard** way, at least in the past thirty and more years! 😄

**Comment by lb90** [comment]
Ah, maybe you mean that by typing text using AltGr the user may inadvertently trigger keyboard shortucts?

**Comment by lmattcris** [comment]
@lb90 Waving 🏳️: I'm following the issue just cause it's related to an application I use. I admit having little knowledge of accelerators under GTK3, sorry!

**Comment by lb90** [comment]
Well, you made a good argument, though! :+1:

@giuspen could you open an issue in https://gitlab.gnome.org/GNOME/gtk/-/issues?

**Comment by giuspen** [comment]
Thanks to everyone that took the time to follow this issue, I raised now the ticket https://gitlab.gnome.org/GNOME/gtk/-/issues/5231
